metis-core/metis-core-common/src/main/java/eu/europeana/metis/core/workflow/plugins/AbstractMetisPlugin.java

```diff
@@ -275,47 +275,51 @@ DpsTask createDpsTaskForHarvestPlugin(Map<String, String> extraParameters, Strin
     if (extraParameters != null) {
       parameters.putAll(extraParameters);
     }
-    parameters.put("PROVIDER_ID", ecloudProvider);
+    parameters.put("PROVIDER_ID", ecloudBasePluginParameters.getEcloudProvider());
     parameters.put("OUTPUT_DATA_SETS",
-        String.format(CommonStringValues.S_DATA_PROVIDERS_S_DATA_SETS_S_TEMPLATE, ecloudBaseUrl,
-            ecloudProvider, ecloudDataset));
+        String.format(CommonStringValues.S_DATA_PROVIDERS_S_DATA_SETS_S_TEMPLATE,
+            ecloudBasePluginParameters.getEcloudBaseUrl(), ecloudBasePluginParameters.getEcloudProvider(),
+            ecloudBasePluginParameters.getEcloudDataset()));
     parameters.put("NEW_REPRESENTATION_NAME", getRepresentationName());
     dpsTask.setParameters(parameters);
 
-    dpsTask.setOutputRevision(createOutputRevisionForExecution(ecloudProvider, false));
+    dpsTask.setOutputRevision(
+        createOutputRevisionForExecution(ecloudBasePluginParameters.getEcloudProvider(), false));
     return dpsTask;
   }
 
-  DpsTask createDpsTaskForProcessPlugin(Map<String, String> extraParameters, String ecloudBaseUrl,
-      String ecloudProvider, String ecloudDataset) {
+  DpsTask createDpsTaskForProcessPlugin(EcloudBasePluginParameters ecloudBasePluginParameters,
+      Map<String, String> extraParameters) {
     Map<String, String> parameters = new HashMap<>();
     if (extraParameters != null) {
       parameters.putAll(extraParameters);
     }
     parameters.put("REPRESENTATION_NAME", getRepresentationName());
     parameters.put("REVISION_NAME", getPluginMetadata().getRevisionNamePreviousPlugin());
-    parameters.put("REVISION_PROVIDER", ecloudProvider);
+    parameters.put("REVISION_PROVIDER", ecloudBasePluginParameters.getEcloudProvider());
     DateFormat dateFormat = new SimpleDateFormat(CommonStringValues.DATE_FORMAT, Locale.US);
     parameters.put("REVISION_TIMESTAMP",
         dateFormat.format(getPluginMetadata().getRevisionTimestampPreviousPlugin()));
+    parameters.put("PREVIOUS_TASK_ID", ecloudBasePluginParameters.getPreviousExternalTaskId());
     parameters.put("NEW_REPRESENTATION_NAME", getRepresentationName());
     parameters.put("OUTPUT_DATA_SETS",
         String.format(CommonStringValues.S_DATA_PROVIDERS_S_DATA_SETS_S_TEMPLATE,
-            ecloudBaseUrl, ecloudProvider, ecloudDataset));
-    return createDpsTaskForPluginWithExistingDataset(parameters, ecloudBaseUrl, ecloudProvider,
-        ecloudDataset, false);
+            ecloudBasePluginParameters.getEcloudBaseUrl(), ecloudBasePluginParameters.getEcloudProvider(),
+            ecloudBasePluginParameters.getEcloudDataset()));
+    return createDpsTaskForPluginWithExistingDataset(parameters,
```

Comment:
Just send the whole object to this private method instead of the three fields?

Reply:
Fixed.